contracts/Vesting.sol

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/// @title Vesting
+/// @author Eldar Dryga
+/// @notice this contract allows you to distribute tokens to users within a certain time
+
+contract Vesting is Ownable, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+    IERC20 private immutable _token;
+
+    uint32 public periodOfDistributionInDays;
+    uint32 public amountOfDistributionPerUser;
+    uint128 public users;
+    uint256 public maxUsers;
+
+    struct UsersInfo {
+        uint256 userTime;
```

Comment:
Maybe not "userTime" but "startTime"?

Reply:
No, this variable changes every time after the "claimRevard" is called, so it would be incorrect